src/com/palmergames/bukkit/towny/object/Town.java

```diff
@@ -36,67 +31,46 @@
-	private List<Resident> outlaws = new ArrayList<>();
+	private final List<Resident> residents = new ArrayList<>();
+	private final List<Resident> outlaws = new ArrayList<>();
 	private List<Location> outpostSpawns = new ArrayList<>();
-	private List<Location> jailSpawns = new ArrayList<>();
+	private final List<Location> jailSpawns = new ArrayList<>();
 	private HashMap<String, PlotGroup> plotGroups = null;
 	
 	private Resident mayor;
 	private int bonusBlocks = 0;
 	private int purchasedBlocks = 0;
-	private double taxes = TownySettings.getTownDefaultTax();
 	private double plotTax= TownySettings.getTownDefaultPlotTax();
 	private double commercialPlotTax = TownySettings.getTownDefaultShopTax();
 	private double plotPrice = 0.0;
 	private double embassyPlotTax = TownySettings.getTownDefaultEmbassyTax();
 	private double maxPercentTaxAmount = TownySettings.getMaxTownTaxPercentAmount();
-	private double commercialPlotPrice, embassyPlotPrice, spawnCost;
+	private double commercialPlotPrice, embassyPlotPrice;
 	private Nation nation;
 	private boolean hasUpkeep = true;
-	private boolean isPublic = TownySettings.getTownDefaultPublic();
 	private boolean isTaxPercentage = TownySettings.getTownDefaultTaxPercentage();
-	private boolean isOpen = TownySettings.getTownDefaultOpen();
-	private String townBoard = "/town set board [msg]";
-	private String tag = "";
 	private TownBlock homeBlock;
 	private TownyWorld world;
-	private Location spawn;
 	private boolean adminDisabledPVP = false; // This is a special setting to make a town ignore All PVP settings and keep PVP disabled.
 	private boolean adminEnabledPVP = false; // This is a special setting to make a town ignore All PVP settings and keep PVP enabled. Overrides the admin disabled too.
 	private UUID uuid;
-	private long registered;
-	private transient List<Invite> receivedinvites = new ArrayList<>();
-	private transient List<Invite> sentinvites = new ArrayList<>();
 	private boolean isConquered = false;
 	private int conqueredDays;
-	private EconomyAccount account;
-	private ConcurrentHashMap<WorldCoord, TownBlock> townBlocks = new ConcurrentHashMap<>();
-	private TownyPermission permissions = new TownyPermission();
+	private final ConcurrentHashMap<WorldCoord, TownBlock> townBlocks = new ConcurrentHashMap<>();
+	private final TownyPermission permissions = new TownyPermission();
 
 	public Town(String name) {
 		super(name);
 		permissions.loadDefault(this);
 	}
 
-	/*
-	 * Not used but required to Implement TownBlockOwner (non-Javadoc)
-	 * @see com.palmergames.bukkit.towny.object.TownBlockOwner#setTownblocks(java.util.List)
-	 */
-	@Override
-	public void setTownblocks(List<TownBlock> townblocks) {
-		//this.townBlocks = townblocks;
-	}
-
 	@Override
 	public List<TownBlock> getTownBlocks() {
-		List<TownBlock> townBlockList = new ArrayList<>();
-		townBlockList.addAll(townBlocks.values());
-		return townBlockList;
+		return Collections.unmodifiableList(new ArrayList<>(townBlocks.values()));
```

Comment:
At this point it might just be more valid to abstract the method to return a `Collection<TownBlock>`